provider/resources.go

```diff
@@ -78,39 +47,26 @@ func Provider() tfbridge.ProviderInfo {
 
 	// Create a Pulumi provider mapping
 	prov := tfbridge.ProviderInfo{
-		P:                    p,
-		Name:                 "tailscale",
-		DisplayName:          "Tailscale",
-		Description:          "A Pulumi package for creating and managing Tailscale cloud resources.",
-		Keywords:             []string{"pulumi", "tailscale"},
-		License:              "Apache-2.0",
-		Homepage:             "https://pulumi.io",
-		GitHubOrg:            "tailscale",
-		Repository:           "https://github.com/pulumi/pulumi-tailscale",
-		Config:               map[string]*tfbridge.SchemaInfo{},
-		PreConfigureCallback: preConfigureCallback,
+		P:            p,
+		Name:         "tailscale",
+		DisplayName:  "Tailscale",
+		Description:  "A Pulumi package for creating and managing Tailscale cloud resources.",
+		Keywords:     []string{"pulumi", "tailscale"},
+		License:      "Apache-2.0",
+		Homepage:     "https://pulumi.io",
+		GitHubOrg:    "tailscale",
+		Repository:   "https://github.com/pulumi/pulumi-tailscale",
+		Config:       map[string]*tfbridge.SchemaInfo{},
+		MetadataInfo: tfbridge.NewProviderMetadata(metadata),
 		Resources: map[string]*tfbridge.ResourceInfo{
 			"tailscale_acl": {
-				Tok: makeResource(mainMod, "Acl"),
 				Fields: map[string]*tfbridge.SchemaInfo{
-					"acl": {
-						CSharpName: "AclJson",
-					},
+					"acl": {CSharpName: "AclJson"},
 				},
 			},
-			"tailscale_device_subnet_routes": {Tok: makeResource(mainMod, "DeviceSubnetRoutes")},
-			"tailscale_dns_nameservers":      {Tok: makeResource(mainMod, "DnsNameservers")},
-			"tailscale_dns_preferences":      {Tok: makeResource(mainMod, "DnsPreferences")},
-			"tailscale_dns_search_paths":     {Tok: makeResource(mainMod, "DnsSearchPaths")},
-			"tailscale_device_authorization": {Tok: makeResource(mainMod, "DeviceAuthorization")},
-			"tailscale_device_tags":          {Tok: makeResource(mainMod, "DeviceTags")},
-			"tailscale_tailnet_key":          {Tok: makeResource(mainMod, "TailnetKey")},
-			"tailscale_device_key":           {Tok: makeResource(mainMod, "DeviceKey")},
 		},
 		DataSources: map[string]*tfbridge.DataSourceInfo{
-			"tailscale_device":  {Tok: makeDataSource(mainMod, "getDevice")},
-			"tailscale_devices": {Tok: makeDataSource(mainMod, "getDevices")},
-			"tailscale_4via6":   {Tok: makeDataSource(mainMod, "get4Via6")},
+			"tailscale_4via6": {Tok: tfbridge.MakeDataSource(mainPkg, mainMod, "get4Via6")},
```

Comment:
do we want to add a comment here that the token computation isn't handling this as expected due to the 4, and therefore it must remain explicit?

Reply:
Sure